smartsim/_core/control/manifest.py

```diff
@@ -126,58 +128,54 @@ def _check_entity_lists_nonempty(self) -> None:
                 raise ValueError(f"{entity_list.name} is empty. Nothing to launch.")
 
     def __str__(self) -> str:
-        s = ""
+        output = ""
         e_header = "=== Ensembles ===\n"
         m_header = "=== Models ===\n"
         db_header = "=== Database ===\n"
         if self.ensembles:
-            s += e_header
+            output += e_header
 
             all_ensembles = self.ensembles
             for ensemble in all_ensembles:
-                s += f"{ensemble.name}\n"
-                s += f"Members: {len(ensemble)}\n"
-                s += f"Batch Launch: {ensemble.batch}\n"
+                output += f"{ensemble.name}\n"
+                output += f"Members: {len(ensemble)}\n"
+                output += f"Batch Launch: {ensemble.batch}\n"
                 if ensemble.batch:
-                    s += f"{str(ensemble.batch_settings)}\n"
-            s += "\n"
+                    output += f"{str(ensemble.batch_settings)}\n"
+            output += "\n"
 
         if self.models:
-            s += m_header
+            output += m_header
             for model in self.models:
-                s += f"{model.name}\n"
+                output += f"{model.name}\n"
                 if model.batch_settings:
-                    s += f"{model.batch_settings}\n"
-                s += f"{model.run_settings}\n"
+                    output += f"{model.batch_settings}\n"
+                output += f"{model.run_settings}\n"
                 if model.params:
-                    s += f"Parameters: \n{fmt_dict(model.params)}\n"
-            s += "\n"
+                    output += f"Parameters: \n{fmt_dict(model.params)}\n"
+            output += "\n"
 
         if self.db:
-            s += db_header
-            s += f"Shards: {self.db.num_shards}\n"
-            s += f"Port: {str(self.db.ports[0])}\n"
-            s += f"Network: {self.db._interfaces}\n"
-            s += f"Batch Launch: {self.db.batch}\n"
+            output += db_header
+            output += f"Shards: {self.db.num_shards}\n"
+            output += f"Port: {str(self.db.ports[0])}\n"
+            output += f"Network: {self.db._interfaces}\n"
+            output += f"Batch Launch: {self.db.batch}\n"
             if self.db.batch:
-                s += f"{str(self.db.batch_settings)}\n"
+                output += f"{str(self.db.batch_settings)}\n"
 
-        s += "\n"
-        return s
+        output += "\n"
+        return output
 
     @property
     def has_db_objects(self) -> bool:
         """Check if any entity has DBObjects to set"""
 
-        def has_db_models(entity: t.Union[EntityList, SmartSimEntity]) -> bool:
-            if hasattr(entity, "_db_models"):
-                return len(entity._db_models) > 0
-            return False
+        def has_db_models(entity: t.Union[EntityList, Model]) -> bool:
+            return len(entity.db_models) > 0
```

Comment:
I think it's a combination of two factors.

First, we've added typehints which should tell users "do it this way" and gives them `make mypy` to find out if they're wrong before they run code.

Second, the main use cases now run through "strongly typed" accessors. For example:

```
@property
def models(self) -> t.List[Model]:
    """Return Model instances in Manifest

    :return: model instances
    :rtype: List[Model]
    """
    _models: t.List[Model] = [
        item for item in self._deployables if isinstance(item, Model)
    ]
    return _models
```
and
```
has_db_objects |= any(
    has_db_models(model) | has_db_scripts(model) for model in self.models
)

```

Since our code no longer passes every item in the collection and instead only sends the models, adding another check is redundant.

The old code did something like:

```
has_dbscripts = any(has_db_objects(x) for x in self._deployables)
```
which made it required that we check the type in each of the has_xxx methods.